android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboardJSHandler.java

```diff
@@ -286,12 +286,13 @@ private static void performLeftDeletions(InputConnection ic, int dn) {
     // subSequence indices are start(inclusive) to end(exclusive)
     CharSequence expectedChars = charsBackup.subSequence(0, charsBackup.length() - (dn + numPairs));
     ic.deleteSurroundingText(dn + numPairs, 0);
-    CharSequence newContext = getCharacterSequence(ic, originalBufferLength - 2*dn);
+    // Shorten the retrieved context by exactly as many characters as were just deleted.
+    CharSequence newContext = getCharacterSequence(ic, originalBufferLength - dn - numPairs);
```

Comment:
One of two changes sufficient to fix the issue (noted before regression testing).